### ISSUE TITLE
allow accounts[0] to be undefined

### ISSUE THREAD

**Issue by dkent600** [daostack/arc.js_legacy#236]
Allow accounts[0] to be undefined.  Issue a logged warning instead of throwing an exception.

This allows for cases where arc.js is used in read-only situations.

When undefined, don't set a default user in `Utils.requireContract`.

**Comment by dkent600** [comment]
A possible corollary is to provide some means of reinitializing `WrapperService` when the default user account changes.

**Comment by dkent600** [comment]
@tibetsprague

**Comment by tibetsprague** [comment]
So then Alchemy would have to check if the default account exists or not?

**Comment by tibetsprague** [comment]
would love to have arc.js auto detect account changes

**Comment by dkent600** [comment]
@tibetsprague 

Yes, the application (Alchemy) would need to check if the default account exists.  No more exception in that case.

I will create a new ticket on creating a Pub/Sub event for when the account changes.

**Comment by dkent600** [comment]
When the user finally sets the default account, the application will have to re-invoke InitializeArcJs
